subiquity/server/controllers/filesystem.py

```diff
@@ -743,7 +764,9 @@ def start_guided_resize(
         gap = gaps.after(disk, partition.offset)
         if gap is None:
             pgs = gaps.parts_and_gaps(disk)
-            raise Exception(f"gap not found after resize, pgs={pgs}")
+            raise StorageConstraintViolationError(
+                f"gap not found after resize, pgs={pgs}"
```

Comment:
I suggest moving this "pgs" bit to a debug log statement, the discussion about "gaps" is going to be kind of inscrutable to an end user and not at all clear what they should do about it.  The case here is we're doing guided resize and the gap we expected to find after resize just isn't there, perhaps a different resize would work better?  We could say some sort of "try again with a different value" type message, maybe.

This is hopefully a case that never is hit!